src/lambdapi.ml

```diff
@@ -111,6 +109,30 @@ let lsp_server_cmd : config -> bool -> string -> unit =
   let run _ = init cfg; Lsp.Lp_lsp.main standard_lsp lsp_log_file in
   Console.handle_exceptions run
 
+(** Printing a decision tree. *)
+let decision_tree_cmd : config -> string -> unit = fun cfg qsym ->
+  let run _ =
+    init cfg;
+    let (mp, sym) =
+      let rec get red mp =
+        match red with
+        | [sym] -> (mp, sym)
+        | m::ms -> get ms (m::mp)
+        | []    -> assert false
+      in
+      let (mp, sym) = get (String.split_on_char '.' qsym) [] in
+      (List.rev mp, sym)
+    in
```

Comment:
This should move to argument parsing (see other comment), but if you have the guarantee that the list of items is non-empty I would just do something like:
```ocaml
let (mp,s) = match List.rev path with [] -> assert false | s::mp -> (List.rev mp, s) in
```